core/runtime/src/main/java/io/quarkus/runtime/ThreadsConfig.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.runtime;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+/**
+ * The core thread pool config. This thread pool is responsible for running
+ * all blocking tasks.
+ */
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public class ThreadsConfig {
+
+    /**
+     * The core thread pool size. This number of threads will always be kept alive.
+     */
+    @ConfigItem(defaultValue = "1")
+    public int coreThreads;
+
+    /**
+     * The maximum number of threads. If this is not specified or <= to zero then
+     * it will be automatically sized to 4 * the number of available processors
+     */
+    @ConfigItem(defaultValue = "0")
+    public int maxThreads;
+
+    /**
+     * The queue size
+     */
+    @ConfigItem(defaultValue = "0")
+    public int queueSize;
+
+    /**
+     * The queue size. For most applications this should be unbounded
+     */
+    @ConfigItem(defaultValue = "0")
+    public int growthResistance;
+
+    /**
+     * The shutdown timeout in milliseconds. Defaults to 10s. If all pending work has not been completed by this time
+     * then additional threads will be spawned to attempt to finish any pending tasks, and the shutdown process will
+     * continue
+     */
+    @ConfigItem(defaultValue = "10000")
+    public long shutdownTimeout;
```

Comment:
These should both be Durations.

Reply:
Do we have a converter for duration?

Reply:
Yeah it's what we're using for all timeout things.